src/api/js/src/high-level/types.ts

```diff
@@ -1,59 +1,97 @@
 import {
-  Z3_ast,
-  Z3_ast_map,
-  Z3_ast_vector,
-  Z3_context,
-  Z3_decl_kind,
-  Z3_func_decl,
-  Z3_func_interp,
-  Z3_model,
-  Z3_probe,
-  Z3_solver,
-  Z3_sort,
-  Z3_sort_kind,
-  Z3_symbol,
-  Z3_tactic,
+    Z3_ast,
+    Z3_ast_map,
+    Z3_ast_vector,
+    Z3_context,
+    Z3_decl_kind,
+    Z3_func_decl,
+    Z3_func_interp,
+    Z3_model, Z3_pattern,
+    Z3_probe,
+    Z3_solver,
+    Z3_sort,
+    Z3_sort_kind,
+    Z3_symbol,
+    Z3_tactic,
 } from '../low-level';
 
 /** @hidden */
 export type AnySort<Name extends string = 'main'> =
-  | Sort<Name>
-  | BoolSort<Name>
-  | ArithSort<Name>
-  | BitVecSort<number, Name>;
+    | Sort<Name>
+    | BoolSort<Name>
+    | ArithSort<Name>
+    | BitVecSort<number, Name>
+    | SMTArraySort<Name>;
 /** @hidden */
 export type AnyExpr<Name extends string = 'main'> =
-  | Expr<Name>
-  | Bool<Name>
-  | Arith<Name>
-  | IntNum<Name>
-  | RatNum<Name>
-  | BitVec<number, Name>
-  | BitVecNum<number, Name>;
+    | Expr<Name>
+    | Bool<Name>
+    | Arith<Name>
+    | IntNum<Name>
+    | RatNum<Name>
+    | BitVec<number, Name>
+    | BitVecNum<number, Name>
+    | SMTArray<Name>;
 /** @hidden */
 export type AnyAst<Name extends string = 'main'> = AnyExpr<Name> | AnySort<Name> | FuncDecl<Name>;
 
 /** @hidden */
-export type SortToExprMap<S extends AnySort<Name>, Name extends string = 'main'> = S extends BoolSort
-  ? Bool<Name>
-  : S extends ArithSort<Name>
-  ? Arith<Name>
-  : S extends BitVecSort<infer Size, Name>
-  ? BitVec<Size, Name>
-  : S extends Sort<Name>
-  ? Expr<Name>
-  : never;
+export type SortToExprMap<S extends AnySort<Name>, Name extends string = 'main'> =
+    S extends BoolSort
+    ? Bool<Name>
+    : S extends ArithSort<Name>
+    ? Arith<Name>
+    : S extends BitVecSort<infer Size, Name>
+    ? BitVec<Size, Name>
+    : S extends SMTArraySort<Name, infer DomainSort, infer RangeSort>
+    ? SMTArray<Name, DomainSort, RangeSort>
+    : S extends Sort<Name>
+    ? Expr<Name, S, Z3_ast>
+    : never;
 
 /** @hidden */
-export type CoercibleToExprMap<S extends CoercibleToExpr<Name>, Name extends string = 'main'> = S extends bigint
-  ? IntNum<Name>
-  : S extends number | CoercibleRational
-  ? RatNum<Name>
-  : S extends boolean
-  ? Bool<Name>
-  : S extends Expr<Name>
-  ? S
-  : never;
+export type CoercibleToExprMap<S extends CoercibleToExpr<Name>, Name extends string = 'main'> =
+    S extends bigint
```

Comment:
the style is inconsistent with the style used on line 39-50 and elsewhere. I guess an editor setting.